### PR TITLE
Homepage back

### PR DESCRIPTION
**Pull request by alejandrobodichon**
### Summary
The news are loaded in the background from a REST server. A endlessScrollListener has been added in the recyclerview.
### Screenshot

image

### Trello card
https://trello.com/c/arumdQt3/11-u5-home-page-backend-news

### CODE REVIEW

File: app/src/main/java/ar/com/wolox/android/example/TrainingApplication.java [hunk 1]
```diff
@@ -22,7 +22,11 @@
 
     @Override
     public void onInit() {
-        Fresco.initialize(this);
+        ImagePipelineConfig config = ImagePipelineConfig.newBuilder(this)
```
Comment (lmiotti) [inline app/src/main/java/ar/com/wolox/android/example/TrainingApplication.java]: What is this?

File: app/src/main/java/ar/com/wolox/android/example/ui/training/homePage/news/EndlessRecyclerViewScrollListener.kt [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    private var loading = true
+    private val startingPageIndex = 0
+
+    private var mLayoutManager: RecyclerView.LayoutManager? = null
+
+    constructor(visibleThreshold: Int, layoutManager: LinearLayoutManager): this(visibleThreshold) {
+        this.mLayoutManager = layoutManager
+    }
+
+    constructor(visibleThreshold: Int, layoutManager: GridLayoutManager): this(visibleThreshold) {
```
Comment (lmiotti) [inline app/src/main/java/ar/com/wolox/android/example/ui/training/homePage/news/EndlessRecyclerViewScrollListener.kt]: We can delete the constructors that you are not using

File: app/src/main/java/ar/com/wolox/android/example/ui/training/login/LoginFragment.java [hunk 3]
```diff
@@ -117,4 +117,5 @@ public void showProgressBar() {
     public void hideProgressBar() {
         mProgressBar.setVisibility(getView().GONE);
     }
+
```
Comment (lmiotti) [inline app/src/main/java/ar/com/wolox/android/example/ui/training/login/LoginFragment.java]: Erase this empty line

File: app/src/main/res/layout/recycler_news.xml [hunk 4]
```diff
@@ -11,6 +12,11 @@
         android:layout_width="@dimen/icon_size"
         android:layout_height="@dimen/icon_size"
         android:layout_marginLeft="@dimen/spacing_large"
+        fresco:actualImageScaleType="centerCrop"
+        fresco:fadeDuration="500"
```
Comment (lmiotti) [inline app/src/main/res/layout/recycler_news.xml]: Put this on integer file

File: app/src/main/res/layout/recycler_news.xml [hunk 5]
```diff
@@ -11,6 +12,11 @@
...
+        fresco:failureImageScaleType="center"
+        fresco:placeholderImageScaleType="centerCrop"
+        fresco:progressBarAutoRotateInterval="1000"
```
Comment (lmiotti) [inline app/src/main/res/layout/recycler_news.xml]: same as above

File: app/src/main/java/ar/com/wolox/android/example/ui/training/homePage/news/NewsPresenter.kt [hunk 6]
```diff
@@ -1,36 +1,79 @@
...
     private var count: Int = 3
+    val PAGE_SIZE = 10
+    var firstTime = true
 
     override fun onViewAttached() {
         super.onViewAttached()
         loadNews()
-        view.showNews(news)
     }
     fun loadNews(): MutableList<News> {
```
Comment (lmiotti) [inline app/src/main/java/ar/com/wolox/android/example/ui/training/homePage/news/NewsPresenter.kt]: Add an empty line above

Comment (lmiotti) [reply app/src/main/java/ar/com/wolox/android/example/ui/training/homePage/news/NewsPresenter.kt]: As discussed on private, follow the MVP model

**Review by lmiotti — CHANGES_REQUESTED** [review]
Few changed to be done and we finish  training. Come on!

**Review by lmiotti — APPROVED** [review]
You have finished training!!! Congratulations!

giphy

**Review by dylan-muszel — APPROVED** [review]
You have finished the training, congratulations! Good luck with the project!

**Pull request merged**